descwl_shear_sims/sim/sim.py

```diff
@@ -334,6 +341,342 @@ def make_seobs(
     )
 
 
+def make_dmsim(
+    *,
+    rng,
+    galaxy_catalog,
+    coadd_dim,
+    g1,
+    g2,
+    psf,
+    star_catalog=None,
+    psf_dim=51,
+    dither=False,
+    rotate=False,
+    bands=['i'],
+    epochs_per_band=1,
+    noise_factor=1.0,
+    cosmic_rays=False,
+    bad_columns=False,
+    star_bleeds=False,
+):
+    """
+    Make simulation data
+
+    Parameters
+    ----------
+    rng: numpy.random.RandomState
+        Numpy random state
+    galaxy_catalog: catalog
+        E.g. WLDeblendGalaxyCatalog or FixedGalaxyCatalog
+    coadd_dim: int
+        Default 351
+    g1: float
+        Shear g1 for galaxies
+    g2: float
+        Shear g2 for galaxies
+    psf: GSObject or PowerSpectrumPSF
+        The psf object or power spectrum psf
+    star_catalog: catalog
+        e.g. StarCatalog
+    psf_dim: int, optional
+        Dimensions of psf image.  Default 51
+    dither: bool, optional
+        Whether to dither the images at the pixel level, default False
+    rotate: bool, optional
+        Whether to rotate the images randomly, default False
+    bands: list, optional
+        Default ['i']
+    epochs_per_band: int, optional
+        Number of epochs per band
+    noise_factor: float, optional
+        Factor by which to multiply the noise, default 1
+    cosmic_rays: bool
+        If True, add cosmic rays
+    bad_columns: bool
+        If True, add bad columns
+    """
+
+    se_dim = get_se_dim(coadd_dim=coadd_dim)
+
+    band_data = {}
+    for band in bands:
+
+        survey = get_survey(gal_type=galaxy_catalog.gal_type, band=band)
+        noise_per_epoch = survey.noise*np.sqrt(epochs_per_band)*noise_factor
+
+        # go down to coadd depth in this band, not dividing by sqrt(nbands)
+        # but we could if we want to be more conservative
+        mask_threshold = survey.noise*noise_factor
+
+        objlist, shifts = galaxy_catalog.get_objlist(
+            survey=survey,
+            g1=g1,
+            g2=g2,
+        )
+        objlist, shifts, bright_objlist, bright_shifts, bright_mags = get_objlist(
+            galaxy_catalog=galaxy_catalog,
+            survey=survey,
+            g1=g1,
+            g2=g2,
+            star_catalog=star_catalog,
+            noise=noise_per_epoch,
+        )
```

Comment:
What is going on with these two calls? It appears the first versions of shifts and objlist are being discarded?

Reply:
looks like a bug, forgot to remove an old call